kroxylicious-integration-tests/src/test/java/io/kroxylicious/kms/service/KmsIT.java

```diff
@@ -0,0 +1,164 @@
+import java.nio.ByteBuffer;
+import java.time.Duration;
+
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.TestTemplate;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * Abstract Kms IT.
+ *
+ * @param <C> The config type
+ * @param <K> The key reference
+ * @param <E> The type of encrypted DEK
+ */
+@SuppressWarnings("java:S5960") // this is test code, it exists in the main module to facilitate its use by concrete test cases
+
+@ExtendWith(RecordEncryptionTestInvocationContextProvider.class)
+class KmsIT<C, K, E> {
+
+    private Kms<K, E> kms;
+
+    private TestKekManager manager;
+
+    KmsIT() {
+    }
+
+    @BeforeEach
+    void beforeEach(TestKmsFacade<?, ?, ?> facade) {
+        facade.start();
+        try {
+            var service = (KmsService<C, K, E>) facade.getKmsServiceClass().getDeclaredConstructor(new Class[]{}).newInstance();
+            kms = service.buildKms((C) facade.getKmsServiceConfig());
+        }
+        catch (InstantiationException | IllegalAccessException | InvocationTargetException | NoSuchMethodException e) {
+            throw new RuntimeException(e);
+        }
+        manager = facade.getTestKekManager();
+    }
+
+    @TestTemplate
+    void resolveKeyByName() {
+        var alias = "mykey";
+        manager.generateKek(alias);
+        var resolved = kms.resolveAlias(alias);
+        assertThat(resolved)
+                .succeedsWithin(Duration.ofSeconds(5))
+                .isEqualTo(alias);
+    }
+
+    @TestTemplate
+    void resolveWithUnknownKey() {
+        var alias = "unknown";
+        var resolved = kms.resolveAlias(alias);
+        assertThat(resolved)
+                .failsWithin(Duration.ofSeconds(5))
+                .withThrowableThat()
+                .withCauseInstanceOf(UnknownAliasException.class);
+    }
+
+    @TestTemplate
+    void generatedEncryptedDekDecryptsBackToPlain() {
+        var alias = "mykey";
+        manager.generateKek(alias);
+
+        var resolveStage = kms.resolveAlias(alias);
+        assertThat(resolveStage).succeedsWithin(Duration.ofSeconds(5));
+
+        var pairStage = kms.generateDekPair(resolveStage.toCompletableFuture().join());
+        assertThat(pairStage).succeedsWithin(Duration.ofSeconds(5));
+        var pair = pairStage.toCompletableFuture().join();
+
+        var decryptedDekStage = kms.decryptEdek(pair.edek());
+        assertThat(decryptedDekStage)
+                .succeedsWithin(Duration.ofSeconds(5))
+                .matches(sk -> SecretKeyUtils.same((DestroyableRawSecretKey) sk, (DestroyableRawSecretKey) pair.dek()));
+    }
+
+    @TestTemplate
+    void generatedDekPairWithUnknownKey() {
+        var alias = "mykey";
+        var key = createAndResolve(alias);
+
+        manager.deleteKek(alias);
+
+        var pairStage = kms.generateDekPair(key);
```

Comment:
do you see this as doing something different to 
```suggestion
        var pairStage = kms.generateDekPair("neverHeardOf");
```

Reply:
it is different, KekRefs aren't Strings.